src/pages/EditProfile/components/ProfileForm/ProfileForm.tsx

```diff
@@ -0,0 +1,102 @@
+import { zodResolver } from '@hookform/resolvers/zod';
+import { useForm } from 'react-hook-form';
+import FormField from '@/pages/editProfile/components/FormField/FormField.tsx';
+import ProfileImageUpload from '@/pages/editProfile/components/ProfileImageUpload/ProfileImageUpload.tsx';
+import { useFormChanged } from '@/pages/editProfile/hooks/useFormChanged.ts';
+import { profileSchema, ProfileFormValues } from '@/pages/editProfile/schema/profileSchema.ts';
+import BoxButton from '@/shared/components/BoxButton/BoxButton';
+import Text from '@/shared/components/Text/Text';
+import * as styles from './profileForm.css.ts';
+
+interface ProfileFormPropTypes {
+  defaultValues: {
+    nickname: string;
+    phoneNumber: string;
+    name: string;
+    profileImageUrl: string;
+  };
+  onSubmit: (data: ProfileFormValues) => void;
+}
+
+const ProfileForm = ({ defaultValues, onSubmit }: ProfileFormPropTypes) => {
+  const {
+    register,
+    handleSubmit,
+    control,
+    formState: { errors, isValid },
+    watch,
+  } = useForm<ProfileFormValues>({
+    resolver: zodResolver(profileSchema),
+    defaultValues: {
+      nickname: defaultValues.nickname,
+      phoneNumber: defaultValues.phoneNumber,
+      name: defaultValues.name,
+      profileImageUrl: defaultValues.profileImageUrl,
+    },
+    mode: 'onChange',
+  });
+
+  const currentNickname = watch('nickname');
+  const currentName = watch('name');
+  const currentPhoneNumber = watch('phoneNumber');
+  const currentProfileImage = watch('profileImageUrl');
+
+  const isChanged = useFormChanged(
+    [currentNickname, currentPhoneNumber, currentName, currentProfileImage],
```

Comment:
```suggestion
const { nickname, name, phoneNumber, profileImageUrl } = watch();

  const isChanged = useFormChanged(
    [nickname, phoneNumber, name, profileImageUrl],
```
객체로 묶어주면 추후에 필드가 추가되더라도 유지보수 측면에서도 유리하고, 코드도 짧아질 것 같아요